management-api-agent-common/src/main/java/io/k8ssandra/metrics/prometheus/CassandraTasksExports.java

```diff
@@ -0,0 +1,345 @@
+                  sizeToSend.getMetricName(),
+                  sizeToSend.getLabelNames(),
+                  labelValues,
+                  totalSizeToSend);
+
+          totalSizeToSendSamples.add(totalSizeToSendSample);
+
+          Collector.MetricFamilySamples.Sample totalSizeSentSample =
+              new Collector.MetricFamilySamples.Sample(
+                  sizeSent.getMetricName(), sizeSent.getLabelNames(), labelValues, totalSizeSent);
+
+          totalSizeSentSamples.add(totalSizeSentSample);
+        }
+      }
+    }
+
+    // Receive
+    MetricFamilySamples filesToReceiveFamily =
+        new MetricFamilySamples(
+            filesToReceive.getMetricName(), Type.GAUGE, "", totalFilesToReceiveSamples);
+
+    MetricFamilySamples filesReceivedFamily =
+        new MetricFamilySamples(
+            filesReceived.getMetricName(), Type.GAUGE, "", totalFilesReceivedSamples);
+
+    MetricFamilySamples sizeToReceiveFamily =
+        new MetricFamilySamples(
+            sizeToReceive.getMetricName(), Type.GAUGE, "", totalSizeToReceiveSamples);
+
+    MetricFamilySamples sizeReceivedFamily =
+        new MetricFamilySamples(
+            sizeReceived.getMetricName(), Type.GAUGE, "", totalSizeReceivedSamples);
+
+    // Send
+    MetricFamilySamples filesToSendFamily =
+        new MetricFamilySamples(
+            filesToSend.getMetricName(), Type.GAUGE, "", totalFilesToSendSamples);
+
+    MetricFamilySamples filesSentFamily =
+        new MetricFamilySamples(filesSent.getMetricName(), Type.GAUGE, "", totalFilesSentSamples);
+
+    MetricFamilySamples sizeToSendFamily =
+        new MetricFamilySamples(sizeToSend.getMetricName(), Type.GAUGE, "", totalSizeToSendSamples);
+
+    MetricFamilySamples sizeSentFamily =
+        new MetricFamilySamples(sizeSent.getMetricName(), Type.GAUGE, "", totalSizeSentSamples);
+
+    return Lists.newArrayList(
+        filesToReceiveFamily,
+        filesReceivedFamily,
+        sizeToReceiveFamily,
+        sizeReceivedFamily,
+        filesToSendFamily,
+        filesSentFamily,
+        sizeToSendFamily,
+        sizeSentFamily);
+  }
+
+  List<MetricFamilySamples> getCompactionStats() {
+
+    // Cassandra's internal CompactionMetrics are close to what we want, but not exactly.
+    // And we can't access CompactionManager.getMetrics() to get them in 3.11
+    List<Map<String, String>> compactions =
+        ShimLoader.instance.get().getCompactionManager().getCompactions().stream()
+            .filter(
+                c -> {
+                  String taskType = c.get("taskType");
+                  OperationType operationType = OperationType.valueOf(taskType);
+                  return operationType != OperationType.COUNTER_CACHE_SAVE
+                      && operationType != OperationType.KEY_CACHE_SAVE
+                      && operationType != OperationType.ROW_CACHE_SAVE;
+                })
+            .collect(Collectors.toList());
```

Comment:
**Issue:** I'm getting the following exception when running in Cassandra 4.0:

```
java.lang.IllegalArgumentException: No enum constant org.apache.cassandra.db.compaction.OperationType.Compaction
	at java.base/java.lang.Enum.valueOf(Unknown Source)
	at org.apache.cassandra.db.compaction.OperationType.valueOf(OperationType.java:20)
	at io.k8ssandra.metrics.prometheus.CassandraTasksExports.lambda$getCompactionStats$0(CassandraTasksExports.java:288)
	at java.base/java.util.stream.ReferencePipeline$2$1.accept(Unknown Source)
	at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(Unknown Source)
	at java.base/java.util.stream.AbstractPipeline.copyInto(Unknown Source)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(Unknown Source)
	at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(Unknown Source)
	at java.base/java.util.stream.AbstractPipeline.evaluate(Unknown Source)
	at java.base/java.util.stream.ReferencePipeline.collect(Unknown Source)
	at io.k8ssandra.metrics.prometheus.CassandraTasksExports.getCompactionStats(CassandraTasksExports.java:293)
	at io.k8ssandra.metrics.prometheus.CassandraTasksExports.collect(CassandraTasksExports.java:46)
	at io.prometheus.client.Collector.collect(Collector.java:45)
	at io.prometheus.client.CollectorRegistry$MetricFamilySamplesEnumeration.findNextElement(CollectorRegistry.java:204)
	at io.prometheus.client.CollectorRegistry$MetricFamilySamplesEnumeration.nextElement(CollectorRegistry.java:219)
	at io.prometheus.client.CollectorRegistry$MetricFamilySamplesEnumeration.nextElement(CollectorRegistry.java:152)
	at io.prometheus.client.exporter.common.TextFormat.write004(TextFormat.java:71)
	at io.prometheus.client.exporter.common.TextFormat.writeFormat(TextFormat.java:53)
	at io.k8ssandra.metrics.http.NettyServerHandler.channelRead0(NettyServerHandler.java:66)
	at io.k8ssandra.metrics.http.NettyServerHandler.channelRead0(NettyServerHandler.java:30)
	at io.netty.channel.SimpleChannelInboundHandler.channelRead(SimpleChannelInboundHandler.java:99)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357)
	at io.netty.handler.codec.MessageToMessageDecoder.channelRead(MessageToMessageDecoder.java:103)
	at io.netty.handler.codec.MessageToMessageCodec.channelRead(MessageToMessageCodec.java:111)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357)
	at io.netty.channel.CombinedChannelDuplexHandler$DelegatingChannelHandlerContext.fireChannelRead(CombinedChannelDuplexHandler.java:436)
	at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:324)
	at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:296)
	at io.netty.channel.CombinedChannelDuplexHandler.channelRead(CombinedChannelDuplexHandler.java:251)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357)
	at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1410)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365)
	at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:919)
	at io.netty.channel.epoll.AbstractEpollStreamChannel$EpollStreamUnsafe.epollInReady(AbstractEpollStreamChannel.java:795)
	at io.netty.channel.epoll.EpollEventLoop.processReady(EpollEventLoop.java:480)
	at io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:378)
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:989)
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.base/java.lang.Thread.run(Unknown Source)
```

Reply:
Okay, I added .toUpperCase() and a catch clause to prevent this Exception. Should help with 4.0 as the CompactionMetrics calls seem to work reliably only with 4.1 and newer.